src/base_output_updater.cpp

```diff
@@ -234,3 +301,42 @@ void OutputUpdater::update_using_indices(BaseHiddenStates &output_states,
     this->updater->update_selected_output_delta_z(output_states, *this->obs,
                                                   delta_states);
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Noise Output Updater
+////////////////////////////////////////////////////////////////////////////////
+
+NoiseOutputUpdater::NoiseOutputUpdater(const std::string model_device)
```

Comment:
Same here. Is this class for hetesrocesdastic, homocedastics or both? My rule of thumb is that if the code overlapped is not > 50%, we wont to have different classes for each type of noises

Reply:
@lhnguyen102 This is only the heteroscedastic case. How would you do it then? Add an extra parameter to the OutputUpdater function with `type_noise="None","heteros","homos"`?

Reply:
I think of using different methods for each type in `OutputUpdater` class, . In this class class, it will have some functions as follow
```python
.update(...)
.update_using_indices(...)
.update_heteros(...)
.update_homos(...)
```
The reason is that I want to give user the control of which updating function that they want to use. It becomes modular. What do you think?

Reply:
Yes, sounds better. And then in the future also add the indices ones for both heteros and homos.